x-pack/legacy/plugins/task_manager/task.ts

```diff
@@ -244,6 +273,19 @@ export interface ConcreteTaskInstance extends TaskInstance {
    */
   runAt: Date;
 
+  /**
+   * The date and time that this task started execution. This is used to determine
+   * the "real" runAt that ended up running the task. This value is only set
+   * when status is set to "running".
+   */
+  startedAt: Date | null;
+
+  /**
+   * The date and time that this task should re-execute if stuck in "running" / timeout
+   * status. This value is only set when status is set to "running".
+   */
+  retryAt: Date | null;
+
```

Comment:
Too bad @chrisdavies  and I didn't know about mapped types in TS when we worked on this. We probably would have made `TaskInterface` a Partial type of `ConcreteTaskInterface`